packages/react-badge/src/components/Badge/useBadge.ts

```diff
@@ -0,0 +1,30 @@
+import * as React from 'react';
+import { makeMergeProps, resolveShorthandProps } from '@fluentui/react-utils';
+import { useMergedRefs } from '@fluentui/react-hooks';
+import { BadgeProps, BadgeState } from './Badge.types';
+
+/**
+ * Consts listing which props are shorthand props.
+ */
+export const badgeShorthandProps = ['icon'];
+
+const mergeProps = makeMergeProps<BadgeState>({ deepMerge: badgeShorthandProps });
+
+/**
+ * Given user props, returns state and render function for a Badge.
+ */
+export const useBadge = (props: BadgeProps, ref: React.Ref<HTMLElement>, defaultProps?: BadgeProps): BadgeState => {
+  const state = mergeProps(
+    {
+      ref: useMergedRefs(ref, React.useRef(null)),
+      shape: 'circular',
+      size: 'medium',
+      icon: { size: props.size },
```

Comment:
I think the icon size ramp is different from badge. We'd need to map from badge size to the appropriate icon size.

Reply:
For now we don't have converged icons with sizes variations.

Reply:
I don't how this applied, can you please clarify? I checked "Icon" stories and don't see there props for icons:

![image](https://user-images.githubusercontent.com/14183168/108070547-f090e400-7064-11eb-9152-5fc6b44f6c8b.png)
![image](https://user-images.githubusercontent.com/14183168/108070567-f5559800-7064-11eb-929a-a9edeaba3611.png)

Reply:
There's no converged icons for now and the ones there's don't support size prop. I think this is gonna work basically when we have converged icons that support sizes, I think @ling1726 faced similar issue right?

Reply:
```suggestion
```

I suggest to remove this for now at least. Icon scaling can be solved by styles (i.e. direct styling), IDK what will be a path there yet. We can solve it later.